server/src/main/java/io/crate/metadata/shard/ShardReferenceResolver.java

```diff
@@ -53,7 +53,7 @@ public class ShardReferenceResolver implements ReferenceResolver<NestableInput<?
 
     private static final Logger LOGGER = LogManager.getLogger(ShardReferenceResolver.class);
     private static final StaticTableReferenceResolver<ShardRowContext> SHARD_REFERENCE_RESOLVER_DELEGATE =
-        new StaticTableReferenceResolver<>(SysShardsTableInfo.create().expressions());
+        new StaticTableReferenceResolver<>(SysShardsTableInfo.create(List::of).expressions());
```

Comment:
Heads up, I think that there is no side-effect that we pass a no-op `Roles` here, but please also double check, maybe I'm missing something.

Reply:
I think you're right as the shards are already protected by the `SysShardsTableInfo.getRouting()`.
But we should add least add that note to highlight it. Passing an empty list here without any further note looks dangerous and confusing.